### PR TITLE
Add terminal section to settings editor.

### PR DESCRIPTION
**Pull request by AlbertHilb**
This PR adds a terminal section to the settings editor.
Moreover, it removes `fontSize` and `theme` `Terminal` properties in favor of more general `getOption`, `setOption` methods.

Cheers

### CODE REVIEW

File: packages/terminal-extension/src/index.ts [hunk 1]
```diff
@@ -90,7 +93,55 @@ function activate(
...
+      });
+    })
+    .catch((reason: Error) => {
+      console.error(reason.message);
+    });
+
+  addCommands(app, serviceManager, tracker, settingRegistry);
 
   // Add some commands to the application view menu.
   const viewGroup = [
```
[the remark this answers is not in the document]
Comment (AlbertHilb) [reply packages/terminal-extension/src/index.ts]: Done.

File: packages/terminal-extension/schema/plugin.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "fontFamily": {
+      "type": "string"
+    },
+    "fontSize": {
+      "type": "integer",
+      "minimum": 9,
+      "maximum": 72
+    },
+    "lineHeight": {
+      "type": "number"
```
[the remark this answers is not in the document]
Comment (AlbertHilb) [reply packages/terminal-extension/schema/plugin.json]: Done.

File: packages/terminal-extension/src/index.ts [hunk 3]
```diff
@@ -222,15 +274,18 @@ export function addCommands(
...
-          widget.content.fontSize = options.fontSize;
-        });
+      let { fontSize } = Terminal.defaultOptions;
+      if (fontSize < 72) {
+        return settingRegistry
+          .set(plugin.id, 'fontSize', fontSize + 1)
+          .catch(showErrorMessage);
       }
     },
     isEnabled
```
[the remark this answers is not in the document]
Comment (AlbertHilb) [reply packages/terminal-extension/src/index.ts]: Done.

File: packages/terminal-extension/src/index.ts [hunk 4]
```diff
@@ -239,32 +294,27 @@ export function addCommands(
...
-          widget.content.fontSize = options.fontSize;
-        });
+      let { fontSize } = Terminal.defaultOptions;
+      if (fontSize > 9) {
+        return settingRegistry
+          .set(plugin.id, 'fontSize', fontSize - 1)
+          .catch(showErrorMessage);
       }
     },
     isEnabled
```
[the remark this answers is not in the document]
Comment (AlbertHilb) [reply packages/terminal-extension/src/index.ts]: Done.

File: packages/terminal-extension/src/index.ts [hunk 5]
```diff
@@ -239,32 +294,27 @@ export function addCommands(
...
-      });
-      commands.notifyCommandChanged(CommandIDs.toggleTheme);
+      let { theme } = Terminal.defaultOptions;
+      theme = theme === 'dark' ? 'light' : 'dark';
+      return settingRegistry
+        .set(plugin.id, 'theme', theme)
+        .then(() => commands.notifyCommandChanged(CommandIDs.toggleTheme))
+        .catch(showErrorMessage);
     },
     isEnabled
```
[the remark this answers is not in the document]
Comment (AlbertHilb) [reply packages/terminal-extension/src/index.ts]: Done.

**Comment by AlbertHilb** [comment]
Hi @ian-r-rose. All of your comments have been addressed.

**Comment by blink1073** [comment]
Nice!

**Comment by dhirschfeld** [comment]
I've been wanting this exact functionality! :heart:

**Pull request merged**